plugins/kapt3/kapt3-base/src/org/jetbrains/kotlin/kapt3/base/incremental/javacVisitors.kt

```diff
@@ -22,24 +22,32 @@ class MentionedTypesTaskListener(
 ) : TaskListener {
 
     var time = 0L
+    var failureReason: String? = null
+
     override fun started(e: TaskEvent) {
         // do nothing, we just process on finish
     }
 
     override fun finished(e: TaskEvent) {
+        if (failureReason != null) return // stop processing if we failed to analyze some sources
+
         if (e.kind != TaskEvent.Kind.ENTER || cache.isAlreadyProcessed(e.sourceFile.toUri())) return
 
-        val l = System.currentTimeMillis()
-        val compilationUnit = e.compilationUnit
+        try {
+            val l = System.currentTimeMillis()
+            val compilationUnit = e.compilationUnit
 
-        val structure = SourceFileStructure(e.sourceFile.toUri())
+            val structure = SourceFileStructure(e.sourceFile.toUri())
 
-        val treeVisitor = TypeTreeVisitor(elementUtils, trees, compilationUnit, structure)
-        compilationUnit.typeDecls.forEach {
-            it.accept(treeVisitor, Visibility.ABI)
+            val treeVisitor = TypeTreeVisitor(elementUtils, trees, compilationUnit, structure)
+            compilationUnit.typeDecls.forEach {
+                it.accept(treeVisitor, Visibility.ABI)
+            }
+            cache.addSourceStructure(structure)
+            time += System.currentTimeMillis() - l
+        } catch (t: Throwable) {
+            failureReason = "Running non-incrementally because analyzing ${e.sourceFile.toUri()} failed."
```

Comment:
Should we print the exception to a debug log so we can understand what went wrong on a user machine?

Reply:
We can add that to debug log. I did not want to add it to info/warn as it is mostly obscure compiler-infra exception, but adding it to debug makes sense.

Reply:
This is now fixed in https://github.com/JetBrains/kotlin/compare/master...google:prr/gavra/kapt-better-error-message?expand=1 but PR is not updated for some reason.